tests/common/grpc_test.go

```diff
@@ -0,0 +1,210 @@
+			useTCP:                 true,
+			clientURLPattern:       "http://127.0.0.1:${MEMBER_PORT}",
+			expectAuthorityPattern: "127.0.0.1:${MEMBER_PORT}",
+		},
+
+		{
+			name:                   "http://domain[:port]",
+			clientURLPattern:       "http://localhost:${MEMBER_PORT}",
+			expectAuthorityPattern: "localhost:${MEMBER_PORT}",
+		},
+		{
+			name:                   "http://address[:port]",
+			clientURLPattern:       "http://127.0.0.1:${MEMBER_PORT}",
+			expectAuthorityPattern: "127.0.0.1:${MEMBER_PORT}",
+		},
+		{
+			name:                   "https://address[:port]",
+			useTCP:                 true,
+			useTLS:                 true,
+			clientURLPattern:       "https://127.0.0.1:${MEMBER_PORT}",
+			expectAuthorityPattern: "127.0.0.1:${MEMBER_PORT}",
+		},
+		{
+			name:                   "https://domain[:port] insecure",
+			useTLS:                 true,
+			useInsecureTLS:         true,
+			clientURLPattern:       "https://localhost:${MEMBER_PORT}",
+			expectAuthorityPattern: "localhost:${MEMBER_PORT}",
+		},
+		{
+			name:                   "https://address[:port] insecure",
+			useTLS:                 true,
+			useInsecureTLS:         true,
+			clientURLPattern:       "https://127.0.0.1:${MEMBER_PORT}",
+			expectAuthorityPattern: "127.0.0.1:${MEMBER_PORT}",
+		},
+		{
+			name:                   "https://domain[:port]",
+			useTLS:                 true,
+			clientURLPattern:       "https://localhost:${MEMBER_PORT}",
+			expectAuthorityPattern: "localhost:${MEMBER_PORT}",
+		},
+		{
+			name:                   "https://address[:port]",
+			useTLS:                 true,
+			clientURLPattern:       "https://127.0.0.1:${MEMBER_PORT}",
+			expectAuthorityPattern: "127.0.0.1:${MEMBER_PORT}",
+		},
+	}
+	for _, tc := range clusterTestCases() {
+		for _, nc := range tcs {
+			t.Run(tc.name+"/"+nc.name, func(t *testing.T) {
+				ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+				defer cancel()
+				clus := testRunner.NewCluster(ctx, t, config.WithClusterConfig(tc.config))
+				defer clus.Close()
+				cc := testutils.MustClient(clus.Client(tc.ClientOptions))
+
+				testutils.ExecuteUntil(ctx, t, func() {
+					putRequestMethod := "/etcdserverpb.KV/Put"
+					_, err := cc.Put(context.TODO(), "foo", "bar")
+					if err != nil {
+						t.Fatal(err)
+					}
+
+					assertAuthority(t, templateAuthority(t, tc.expectAuthorityPattern, clus.Members[0]), clus, putRequestMethod)
+				})
+			})
+		}
+	}
+}
+
+func templateAuthority(t *testing.T, pattern string, m *integration.Member) string {
+	t.Helper()
+	authority := pattern
+	authority = strings.ReplaceAll(authority, "${MEMBER_PORT}", m.GrpcPortNumber())
+	authority = strings.ReplaceAll(authority, "${MEMBER_NAME}", m.Name)
+	return authority
+}
+
+func assertAuthority(t *testing.T, expectedAuthority string, clus *integration.Cluster, filterMethod string) {
+	t.Helper()
+	requestsFound := 0
+	for _, m := range clus.Members {
+		for _, r := range m.RecordedRequests() {
```

Comment:
Not sure RecordedRequests will work here. You might have to wait for unified logging capture. See https://github.com/etcd-io/etcd/pull/15718#issuecomment-1510613186